log.go

```diff
@@ -64,95 +64,95 @@ func SetMaxLogSize(logSize int64) {
 }
 
-func (log *Log) Panic(message ...interface{}) {
+func (log *Log) Panic(format string, message ...interface{}) {
 	log.rotateLog()
+	log.Println(format, message...)
 
-	log.Println(message)
-	panic(message)
+	panic(fmt.Sprintf(format, message...))
 }
 
-func Panic(message ...interface{}) {
-	std.Panic(message)
+func Panic(format string, message ...interface{}) {
+	std.Panic(format, message...)
 }
 
-func (log *Log) Critical(message ...interface{}) {
+func (log *Log) Critical(format string, message ...interface{}) {
 	log.rotateLog()
 
-	log.Println(message)
+	log.Println(format, message...)
 }
 
-func Critical(message ...interface{}) {
-	std.Critical(message)
+func Critical(format string, message ...interface{}) {
+	std.Critical(format, message...)
 }
 
-func (log *Log) Error(message ...interface{}) {
+func (log *Log) Error(format string, message ...interface{}) {
 	log.rotateLog()
 
-	if log.getLevel() <= ERROR {
-		log.Println(message)
+	if log.level <= ERROR {
+		log.Println(format, message...)
 	}
 }
 
-func Error(message ...interface{}) {
-	std.Error(message)
+func Error(format string, message ...interface{}) {
+	std.Error(format, message...)
 }
 
-func (log *Log) Warn(message ...interface{}) {
+func (log *Log) Warn(format string, message ...interface{}) {
 	log.rotateLog()
 
-	if log.getLevel() <= WARN {
-		log.Println(message)
+	if log.level <= WARN {
+		log.Println(format, message...)
 	}
 }
 
-func Warn(message ...interface{}) {
-	std.Warn(message)
+func Warn(format string, message ...interface{}) {
+	std.Warn(format, message...)
 }
 
-func (log *Log) Info(message ...interface{}) {
+func (log *Log) Info(format string, message ...interface{}) {
 	log.rotateLog()
 
 	if log.getLevel() <= INFO {
-		log.Println(message)
+		log.Println(format, message...)
 	}
 }
 
-func Info(message ...interface{}) {
-	std.Info(message)
+func Info(format string, message ...interface{}) {
+	std.Info(format, message...)
 }
 
-func (log *Log) Debug(message ...interface{}) {
+func (log *Log) Debug(format string, message ...interface{}) {
 	log.rotateLog()
 
 	if log.getLevel() <= DEBUG {
-		log.Println(message)
+		log.Println(format, message...)
 	}
 }
 
-func Debug(message ...interface{}) {
-	std.Debug(message)
+func Debug(format string, message ...interface{}) {
+	std.Debug(format, message...)
 }
 
-func (log *Log) Trace(message ...interface{}) {
+func (log *Log) Trace(format string, message ...interface{}) {
 	log.rotateLog()
 
 	if log.getLevel() <= TRACE {
-		log.Println(message)
+		log.Println(format, message...)
 	}
 }
 
-func Trace(message ...interface{}) {
-	std.Trace(message)
+func Trace(format string, message ...interface{}) {
+	std.Trace(format, message...)
 }
 
-func (log *Log) Println(message ...interface{}) {
-	log.rotateLog()
+func (log *Log) Println(format string, message ...interface{}) {
+	pidStr := fmt.Sprintf("[%d]", os.Getpid())
 
-	internal_logger.Println(message)
+	internal_logger.Printf(pidStr+" "+format, message...)
```

Comment:
Small comment, this might read a little better as:
```
formatWithPid := fmt.Sprintf("[%d] %s", os.Getpid(), format)
internal_logger.Printf(formatWithPid, message...)
```